t/currency.t

```diff
@@ -67,6 +67,19 @@ subtest 'AlphaVantage' => sub {
 };
 
 
+# Check CurrencyFreaks
```

Comment:
 kapatel  ~  git  finance-quote  TEST_CURRENCYFREAKS_API_KEY=<hidden> ONLINE_TEST=1 dzil run
 prove -lv t/currency.t
[DZ] building distribution under .build/W0DOu5c_gB for installation
[DZ] beginning to build Finance-Quote
[Git::NextVersion] Bumping version from 1.61 to 1.62
[DZ] guessing dist's main_module is lib/Finance/Quote.pm
[@Basic/ExtraTests] rewriting author test xt/author/pod-syntax.t
[DZ] writing Finance-Quote in .build/W0DOu5c_gB
[MarkdownInRoot] overriding README.md in root
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Finance::Quote
Writing MYMETA.yml and MYMETA.json
cp lib/Finance/Quote/MorningstarAU.pm blib/lib/Finance/Quote/MorningstarAU.pm
cp lib/Finance/Quote/IEXCloud.pm blib/lib/Finance/Quote/IEXCloud.pm
cp lib/Finance/Quote.pm blib/lib/Finance/Quote.pm
cp lib/Finance/Quote/MorningstarJP.pm blib/lib/Finance/Quote/MorningstarJP.pm
cp lib/Finance/Quote/MorningstarCH.pm blib/lib/Finance/Quote/MorningstarCH.pm
cp lib/Finance/Quote/ASX.pm blib/lib/Finance/Quote/ASX.pm
cp lib/Finance/Quote/Consorsbank.pm blib/lib/Finance/Quote/Consorsbank.pm
cp lib/Finance/Quote/BSEIndia.pm blib/lib/Finance/Quote/BSEIndia.pm
cp lib/Finance/Quote/CurrencyRates/ECB.pm blib/lib/Finance/Quote/CurrencyRates/ECB.pm
cp lib/Finance/Quote/GoogleWeb.pm blib/lib/Finance/Quote/GoogleWeb.pm
cp lib/Finance/Quote/DWS.pm blib/lib/Finance/Quote/DWS.pm
cp lib/Finance/Quote/CSE.pm blib/lib/Finance/Quote/CSE.pm
cp lib/Finance/Quote/ASEGR.pm blib/lib/Finance/Quote/ASEGR.pm
cp lib/Finance/Quote/AEX.pm blib/lib/Finance/Quote/AEX.pm
cp lib/Finance/Quote/HU.pm blib/lib/Finance/Quote/HU.pm
cp lib/Finance/Quote/Currencies.pm blib/lib/Finance/Quote/Currencies.pm
cp lib/Finance/Quote/FTfunds.pm blib/lib/Finance/Quote/FTfunds.pm
cp lib/Finance/Quote/Finanzpartner.pm blib/lib/Finance/Quote/Finanzpartner.pm
cp lib/Finance/Quote/Bourso.pm blib/lib/Finance/Quote/Bourso.pm
cp lib/Finance/Quote/Deka.pm blib/lib/Finance/Quote/Deka.pm
cp lib/Finance/Quote/AlphaVantage.pm blib/lib/Finance/Quote/AlphaVantage.pm
cp lib/Finance/Quote/Fool.pm blib/lib/Finance/Quote/Fool.pm
cp lib/Finance/Quote/BVB.pm blib/lib/Finance/Quote/BVB.pm
cp lib/Finance/Quote/CurrencyRates/Fixer.pm blib/lib/Finance/Quote/CurrencyRates/Fixer.pm
cp lib/Finance/Quote/GoldMoney.pm blib/lib/Finance/Quote/GoldMoney.pm
cp lib/Finance/Quote/MarketWatch.pm blib/lib/Finance/Quote/MarketWatch.pm
cp lib/Finance/Quote/Bloomberg.pm blib/lib/Finance/Quote/Bloomberg.pm
cp lib/Finance/Quote/Fondsweb.pm blib/lib/Finance/Quote/Fondsweb.pm
cp lib/Finance/Quote/IndiaMutual.pm blib/lib/Finance/Quote/IndiaMutual.pm
cp lib/Finance/Quote/CurrencyRates/OpenExchange.pm blib/lib/Finance/Quote/CurrencyRates/OpenExchange.pm
cp lib/Finance/Quote/CurrencyRates/YahooJSON.pm blib/lib/Finance/Quote/CurrencyRates/YahooJSON.pm
cp lib/Finance/Quote/CurrencyRates/CurrencyFreaks.pm blib/lib/Finance/Quote/CurrencyRates/CurrencyFreaks.pm
cp lib/Finance/Quote/CurrencyRates/AlphaVantage.pm blib/lib/Finance/Quote/CurrencyRates/AlphaVantage.pm
cp lib/Finance/Quote/BorsaItaliana.pm blib/lib/Finance/Quote/BorsaItaliana.pm
cp lib/Finance/Quote/Comdirect.pm blib/lib/Finance/Quote/Comdirect.pm
cp lib/Finance/Quote/NZX.pm blib/lib/Finance/Quote/NZX.pm
cp lib/Finance/Quote/Oslobors.pm blib/lib/Finance/Quote/Oslobors.pm
cp lib/Finance/Quote/TMX.pm blib/lib/Finance/Quote/TMX.pm
cp lib/Finance/Quote/SIX.pm blib/lib/Finance/Quote/SIX.pm
cp lib/Finance/Quote/XETRA.pm blib/lib/Finance/Quote/XETRA.pm
cp lib/Finance/Quote/OnVista.pm blib/lib/Finance/Quote/OnVista.pm
cp lib/Finance/Quote/TwelveData.pm blib/lib/Finance/Quote/TwelveData.pm
cp lib/Finance/Quote/YahooWeb.pm blib/lib/Finance/Quote/YahooWeb.pm
cp lib/Finance/Quote/Stooq.pm blib/lib/Finance/Quote/Stooq.pm
cp lib/Finance/Quote/Tradegate.pm blib/lib/Finance/Quote/Tradegate.pm
cp lib/Finance/Quote/NSEIndia.pm blib/lib/Finance/Quote/NSEIndia.pm
cp lib/Finance/Quote/Troweprice.pm blib/lib/Finance/Quote/Troweprice.pm
cp lib/Finance/Quote/MorningstarUK.pm blib/lib/Finance/Quote/MorningstarUK.pm
cp lib/Finance/Quote/TreasuryDirect.pm blib/lib/Finance/Quote/TreasuryDirect.pm
cp lib/Finance/Quote/Sinvestor.pm blib/lib/Finance/Quote/Sinvestor.pm
cp lib/Finance/Quote/Tiaacref.pm blib/lib/Finance/Quote/Tiaacref.pm
cp lib/Finance/Quote/YahooJSON.pm blib/lib/Finance/Quote/YahooJSON.pm
cp lib/Finance/Quote/UserAgent.pm blib/lib/Finance/Quote/UserAgent.pm
cp lib/Finance/Quote/Union.pm blib/lib/Finance/Quote/Union.pm
cp lib/Finance/Quote/ZA.pm blib/lib/Finance/Quote/ZA.pm
cp lib/Finance/Quote/StockData.pm blib/lib/Finance/Quote/StockData.pm
cp lib/Finance/Quote/SEB.pm blib/lib/Finance/Quote/SEB.pm
cp lib/Finance/Quote/TesouroDireto.pm blib/lib/Finance/Quote/TesouroDireto.pm
cp lib/Finance/Quote/TSP.pm blib/lib/Finance/Quote/TSP.pm
t/currency.t ..
1..7
Unknown curreny rates method: DoesNotExist at t/currency.t line 53.
ok 1
# Subtest: AlphaVantage
    1..0 # SKIP Set $ENV{TEST_ALPHAVANTAGE_API_KEY} to run this test; get one at https://www.alphavantage.co
ok 2 # skip Set $ENV{TEST_ALPHAVANTAGE_API_KEY} to run this test; get one at https://www.alphavantage.co
# Subtest: CurrencyFreaks
    1..12
    ok 1
    ok 2 - 100.00 USD -> EUR = 93.0505
    ok 3 # skip identity check because different currencies
    ok 4 - 1.00 GBP -> IDR = 20070.5713931623
    ok 5 # skip identity check because different currencies
    ok 6 - 1.23 IDR -> CAD = 0.000105106013362395
    ok 7 # skip identity check because different currencies
    ok 8 - 10.00 AUD -> AUD = 10.00
    ok 9 - identity check
    ok 10 - 1.0 INR -> INR = 1.0
    ok 11 - identity check
    ok 12 - 20.12 ZZZ -> GBP failed as expected
ok 3 - CurrencyFreaks
# Subtest: ECB
    1..10
    ok 1
    ok 2 - 100.00 USD -> EUR = 93.179276928811
    ok 3 # skip identity check because different currencies
    ok 4 - 1.00 GBP -> IDR = 20057.584743299
    ok 5 # skip identity check because different currencies
    ok 6 - 1.23 IDR -> CAD = 0.000105026054409306
    ok 7 # skip identity check because different currencies
    ok 8 - 10.00 AUD -> AUD = 10.00
    ok 9 - identity check
    ok 10 - 20.12 ZZZ -> GBP failed as expected
ok 4 - ECB
# Subtest: Fixer
    1..0 # SKIP Set $ENV{TEST_FIXER_API_KEY} to run this test; get one at https://fixer.io
ok 5 # skip Set $ENV{TEST_FIXER_API_KEY} to run this test; get one at https://fixer.io
# Subtest: YahooJSON
    1..10
    ok 1
    ok 2 - 100.00 USD -> EUR = 92.72
    ok 3 # skip identity check because different currencies
    ok 4 - 1.00 GBP -> IDR = 20046.45
    ok 5 # skip identity check because different currencies
    ok 6 - 1.23 IDR -> CAD = 0.000104965822104266
    ok 7 # skip identity check because different currencies
    ok 8 - 10.00 AUD -> AUD = 10.00
    ok 9 - identity check
    ok 10 - 20.12 ZZZ -> GBP failed as expected
ok 6 - YahooJSON
# Subtest: Failover
    1..0 # SKIP Set $ENV{TEST_FIXER_API_KEY} to run this test; get one at https://fixer.io
ok 7 # skip Set $ENV{TEST_FIXER_API_KEY} to run this test; get one at https://fixer.io
ok
All tests successful.
Files=1, Tests=7,  4 wallclock secs ( 0.03 usr  0.00 sys +  0.44 cusr  0.03 csys =  0.50 CPU)
Result: PASS
[DZ] all's well; removing .build/W0DOu5c_gB
 kapatel  ~  git  finance-quote  